File: carelog/src/constants/bodyParts.ts
export type BodyPartId =
  | 'HEAD_BRAIN'
  | 'CHEST_HEART'
  | 'ABDOMEN'
  | 'BACK_SPINE'
  | 'ARMS_HANDS'
  | 'LEGS_FEET'
  | 'SKIN'
  | 'GENERAL';

export interface BodyPart {
  id: BodyPartId;
  label: string;
  icon: string;           // MaterialCommunityIcons name
  description: string;    // Sub-label shown on card
  svgRegionId: string;    // Matches SVG path id for highlight
}

export const BODY_PARTS: BodyPart[] = [
  { id: 'HEAD_BRAIN',  label: 'Head & Brain',         icon: 'head-cog-outline',       description: 'Eyes, ears, nose, throat, brain', svgRegionId: 'region-head'    },
  { id: 'CHEST_HEART', label: 'Chest & Heart',        icon: 'heart-pulse',            description: 'Heart, lungs, chest wall',        svgRegionId: 'region-chest'   },
  { id: 'ABDOMEN',     label: 'Abdomen',              icon: 'stomach',                description: 'Stomach, liver, kidneys',         svgRegionId: 'region-abdomen' },
  { id: 'BACK_SPINE',  label: 'Back & Spine',         icon: 'human-handsdown',        description: 'Cervical, lumbar, sacral',        svgRegionId: 'region-back'    },
  { id: 'ARMS_HANDS',  label: 'Arms & Hands',         icon: 'arm-flex-outline',       description: 'Shoulder, elbow, wrist, fingers', svgRegionId: 'region-arms'    },
  { id: 'LEGS_FEET',   label: 'Legs & Feet',          icon: 'shoe-print',             description: 'Hip, knee, ankle, foot',          svgRegionId: 'region-legs'    },
  { id: 'SKIN',        label: 'Skin',                 icon: 'hand-back-right-outline', description: 'Rashes, infections, wounds',      svgRegionId: 'region-skin'    },
  { id: 'GENERAL',     label: 'General / Whole Body', icon: 'human',                  description: 'Fever, fatigue, allergies',       svgRegionId: 'region-general' },
];

File: flutter_version/lib/constants/body_parts.dart
import 'package:flutter/material.dart';

/// Body-part catalog ported from `../carelog/src/constants/bodyParts.ts`.
/// `id` strings are persisted on visits, so they must match the RN app exactly.
/// Icons are the nearest built-in Material `Icons.*` to the RN MDI names
/// (the MDI reference is noted in a comment per row — see PRD §2 icon note).
class BodyPart {
  final String id;
  final String label;
  final IconData icon;
  final String description;

  const BodyPart({
    required this.id,
    required this.label,
    required this.icon,
    required this.description,
  });
}

const List<BodyPart> kBodyParts = [
  BodyPart(
    id: 'HEAD_BRAIN',
    label: 'Head & Brain',
    icon: Icons.psychology_outlined, // mdi: head-cog-outline
    description: 'Eyes, ears, nose, throat, brain',
  ),
  BodyPart(
    id: 'CHEST_HEART',
    label: 'Chest & Heart',
    icon: Icons.monitor_heart_outlined, // mdi: heart-pulse
    description: 'Heart, lungs, chest wall',
  ),
  BodyPart(
    id: 'ABDOMEN',
    label: 'Abdomen',
    icon: Icons.lunch_dining_outlined, // mdi: stomach
    description: 'Stomach, liver, kidneys',
  ),
  BodyPart(
    id: 'BACK_SPINE',
    label: 'Back & Spine',
    icon: Icons.airline_seat_recline_normal, // mdi: human-handsdown
    description: 'Cervical, lumbar, sacral',
  ),
  BodyPart(
    id: 'ARMS_HANDS',
    label: 'Arms & Hands',
    icon: Icons.fitness_center, // mdi: arm-flex-outline
    description: 'Shoulder, elbow, wrist, fingers',
  ),
  BodyPart(
    id: 'LEGS_FEET',
    label: 'Legs & Feet',
    icon: Icons.directions_walk, // mdi: shoe-print
    description: 'Hip, knee, ankle, foot',
  ),
  BodyPart(
    id: 'SKIN',
    label: 'Skin',
    icon: Icons.healing, // mdi: hand-back-right-outline
    description: 'Rashes, infections, wounds',
  ),
  BodyPart(
    id: 'GENERAL',
    label: 'General / Whole Body',
    icon: Icons.accessibility_new, // mdi: human
    description: 'Fever, fatigue, allergies',
  ),
];

/// Lookup by id, or null if unknown.
BodyPart? bodyPartById(String id) {
  for (final bp in kBodyParts) {
    if (bp.id == id) return bp;
  }
  return null;
}
